### ISSUE TITLE
[FR] Add transitions to modules and panels

### ISSUE THREAD

**Issue by Sacules** [darktable-org/darktable#11312]
**Is your feature request related to a problem? Please describe.**
Panels and modules content appear or disappear instantly when you toggle them. This is quite jarring in terms of UX, and a smooth transition here and there goes a long way into making something feel very polished. At least in my experience as a web developer :)

**Describe the solution you'd like**
A simple transition can be easily achieved using GtkRevealer, introduced in GTK 3.10. Not much extra code is needed.

**Alternatives**
I don't know of other methods of achieving this in GTK at the moment.

**Additional context**
I've tried my hand at this with modules, and it was fairly straightforward. [Here's a quick demo](https://imgur.com/6PYLKvj).

I'd like to know if this is of interest for the project - I've been using darktable for almost 2 years now and love it, so I thought I could contribute back with some small UX improvements.

**Comment by paperdigits** [comment]
I think this is cool, so long as the animation stays pretty fast. Some modules have a large height, and I'd not want to wait while it opens/closes.

**Comment by elstoc** [comment]
This looks great when only a single module is expanded at a time, though I don't know how well it will play with other combinations of the scroll/expand preferences:

![Screenshot_2022-03-12_15-29-09](https://user-images.githubusercontent.com/9555491/158024265-f153cc25-ff42-4e3f-b8f0-b0b6464812d5.png)

Just something to bear in mind when you come to develop it.

**Comment by dterrahe** [comment]
Agreed, looks cool in principle for modules. Please also check if it works correctly when activated via shortcuts.

For panels, it might take more resources to make it behave smoothly, since the image in the center of the screen may need to generate multiple zoom levels.

**Comment by Sacules** [comment]
Thanks, I didn't account for those scroll/expand preferences, I'll have to dig a little deeper to make them work smoothly. And yes, panels are proving a challenge for that reason - I'm still new to the DT codebase, where can I see the code for the image resizing? And for the lighttable as well.

So far I've added a transition to the left panel to test things out, and it feels quite pleasing. [Here's a quick demo](https://imgur.com/hdnyL7J) (the arrow is a little buggy though).

**Comment by elstoc** [comment]
Looks good to me. It might be worth trying an image to which you've added some more heavy processing steps (like diffuse/sharpen) so that you can see how the darkroom view copes with redrawing the image when you expand/collapse the panel.

**Comment by dterrahe** [comment]
Just as an FYI; some enhancements were added recently to sliders/comboboxes to handle margins and padding which may make resizing especially the right panel when large modules are open less smooth. So though I think having a transition effect for opening and closing individual modules might look good and add usability by making clearer what is happening, I personally would like to keep opening/closing side panels instantaneous. And make sure you test your changes on the latest master.

**Comment by Sacules** [comment]
That's fair, tbh it's proving hard to make the panels open smoothly without tremendous lag due to the resizing that happens on both the lighttable and the darkroom.

On modules, I've tried to make the "scroll to processing modules when expanded/collapsed" work but the problem is that the scroll happens right away when you toggle a module and the result is quite jarring since it doesn't wait until the transition is over, so it ends up scrolling while the module is opening and it misses the correct scroll position.

**Comment by Sacules** [comment]
Alright, after a lot of trial and error, and digging deeper into Gtk signals and similar, I managed to make it work and even simplify some of the code in the way. [Here's a demo](https://imgur.com/cMFpfkm) with "scroll to processing modules when expanded/collapsed" enabled, works for both the lighttable and the darkroom. I'll leave panels for another ocassion - is it okay if I open a PR with the changes I've made to the modules?
